tokio/src/net/unix/pipe.rs

```diff
@@ -623,6 +695,21 @@ impl Sender {
             .registration()
             .try_io(Interest::WRITABLE, || (&*self.io).write_vectored(buf))
     }
+
+    /// Converts the pipe into an [`OwnedFd`].
+    ///
+    /// This function will deregister this pipe end from the event loop, set
+    /// it in blocking mode and perform the conversion.
+    pub fn into_owned_fd(self) -> io::Result<OwnedFd> {
+        let mio_pipe = self.io.into_inner()?;
+        set_blocking(&mio_pipe)?;
```

Comment:
I don't think this should set the fd into blocking mode.

Reply:
As a side note, we should document the fact that ChildStdout::into_owned_fd and others set the blocking mode.

Reply:
Perhaps split into two methods: `into_blocking_fd`, `into_nonblocking_fd`.